app/models/item.rb

```diff
@@ -1,3 +1,12 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
+  has_many :customer_items
+  has_many :customers, through: :customer_items
+
+  validates :name, presence: true
+  validates :price, presence: true
+
+  def customer_count
+    customers.count
```

Comment:
Nice job here! One thing you could add to make sure that you are getting a unique list of customers is to add a `.distinct` before your `.count`.